### PR TITLE
CoinJoin UI improvements

### PR DESCRIPTION
**Pull request by andrewkozlik**
Updates to the UI as discussed with @hynek-jina. Screens before and after. The last case has two alternatives, the latter with a new warning icon. @hynek-jina please ACK the revised design and say which alternative you prefer for the last one.

image image

image image

image image

image image image

### CODE REVIEW

**Comment by hynek-jina** [comment]
Thanks. Yes.
And for the last screen use the gear please (_I think that the warning might scare someone into impulsively unplugging the Trezor_)

**Comment by andrewkozlik** [comment]
> Maybe even waiting - clock icon or something like that could be better for the last screen.. But the gear is ok as well..

I left the read gear icon, I think it's fine.

File: core/src/apps/bitcoin/common.py [hunk 1]
```diff
@@ -179,3 +180,20 @@ def tagged_hashwriter(tag: bytes) -> HashWriter:
...
+
+
+def format_fee_rate(
+    fee_rate: float, coin: CoinInfo, include_shortcut: bool = False
+) -> str:
+    # Use format_amount to get correct thousands separator -- micropython's built-in
+    # formatting doesn't add thousands sep to floating point numbers.
+    # We multiply by 100 to get a fixed-point integer with two decimal places,
+    # and add 0.5 to round to the nearest integer.
+    fee_rate_formatted = format_amount(int(fee_rate * 100 + 0.5), 2)
```
Comment (mmilata) [inline core/src/apps/bitcoin/common.py:192]: Nit: why not `math.ceil`?

**Review by mmilata — APPROVED** [review]
LGTM, one nit in regarding existing code.

Adding `show_coinjoin` for UI2 to my TODO.

File: core/src/apps/bitcoin/common.py (hunk 1 above)
Comment (prusnak) [reply core/src/apps/bitcoin/common.py:192]: First of all, this is a refactor - the original code already exists in the master branch.

Second of all, `math.ceil` is not what `int(x + 0.5)` does. I guess you meant `round`? 

The original code was added in a41cfadbf1eff82e1e25c1ab7b960e73f1f0cdfa by @marnova. Martin: Did you have any special reason why not to use just `round(fee_rate, 2)`? We don't use this anywhere in the codebase, so there might be a reason why not to use it, but I don't know of any.

Comment (prusnak) [reply core/src/apps/bitcoin/common.py:192]: First of all, this is a refactor - the original code already exists in the master branch.

Second of all, `math.ceil` is not what `int(x + 0.5)` does. I guess you meant `round`? 

The original code was added in 114376f7344b688dca3da65ec0ad9af16935ade8 by @matejcik. Matejcik: Did you have any special reason why not to use just `round(fee_rate, 2)`? We don't use this anywhere in the codebase, so there might be a reason why not to use it, but I don't know of any.

Comment (prusnak) [reply core/src/apps/bitcoin/common.py:192]: Ah, stupid me - the comment above explains it: "Use format_amount to get correct thousands separator  - micropython's built-in formatting doesn't add thousands sep to floating point numbers."

Sorry for the noise.

Comment (andrewkozlik) [reply core/src/apps/bitcoin/common.py:192]: `ceil` would round up, but we want to round to the nearest. Micropython doesn't implement `round()` AFAIK.

Comment (mmilata) [reply core/src/apps/bitcoin/common.py:192]: It does, not in `math` but in `builtins`. Agreed that this is out of scope, apologies.

Comment (andrewkozlik) [reply core/src/apps/bitcoin/common.py:192]: Oh interesting, that's good to know, thanks. Anyway, this particular line of code originated [here](https://github.com/trezor/trezor-firmware/commit/114376f7344b688dca3da65ec0ad9af16935ade8), so I felt no need to change it. Maybe @matejcik can provide insight as to why `int(x+0.5)` is better than `round(x)`, but TBH I think it doesn't matter.

Comment (matejcik) [reply core/src/apps/bitcoin/common.py:192]: i vaguely remember that `round` still returns a float. i might be wrong though

**Pull request merged**